### PR TITLE
[POC] Tweaks for the mission_control package

### PR DESCRIPTION
**Pull request by hidmic**
# Description
Some tweaks and ideas on top of #101. This patch:

- Revamps the `Mission` abstraction
  - Decouples mission and BT states
  - Pushes abort behavior under mission
- Simplifies mission control node implementation
- Fixes CMakeLists.txt
- Fixes style issues

## Type of change

Please mark options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The code builds clean without any errors or warnings
- [ ] Unit tests are passing
- [x] Linter tests are passing

### CODE REVIEW

**Comment by hidmic** [comment]
Tests are not passing yet. Have to flesh out a few things still, but I think this goes in the right direction. I also left TODOs all over the place so we can progressively improve the code.

**Comment by hidmic** [comment]
Alright, got a few more tests passing. Others are failing for good reason:

- We're mishandling optional inputs in all behaviors e.g. a zero roll is a perfectly valid set point. We cannot (nor we have to) use 0 as a sentinel.
- No timeout should mean no timeout, not a zero timeout.
- Tests do not deal well with rapid state changes i.e. if a mission goes from EXECUTING to ABORTING to COMPLETE too quickly, the test may not ever see the intermediate state.

I'll make sure the above's exhaustive. Then, I think we should merge and solve above problems one by one. WDYT @patapufate ?

File: mission_control/src/mission.cpp [hunk 1]
```diff
@@ -36,46 +35,175 @@
...
   }
+  return *this;
 }
+
+Mission& Mission::preempt()
+{
+  switch (status_)
+  {
+    case Mission::Status::PENDING:
+      status_ = Mission::Status::PREEMPTED;
```
Comment (patapufate) [inline mission_control/src/mission.cpp:185]: nit: status are defined as PREEMPTED in all the cases.

**Review by patapufate — APPROVED** [review]
Despites the mission test, LGTM

File: mission_control/src/mission.cpp (hunk 1 above)
Comment (hidmic) [reply mission_control/src/mission.cpp:185]: Not in all cases, but in three (3) states. And for each one of those cases, something special gets done.

I find this easier to read, despite the duplication. What would you propose?

Comment (patapufate) [reply mission_control/src/mission.cpp:185]: +1 for the sake of readability

**Comment by hidmic** [comment]
Alright, let's get this going.

**Pull request merged**